Makefile.dep

```diff
@@ -2,6 +2,13 @@ ifneq (,$(filter libcoap,$(USEPKG)))
     USEMODULE += posix_sockets
 endif
 
+ifneq (,$(filter nhdp,$(USEMODULE)))
+  USEMODULE += conn_udp
+  USEMODULE += vtimer
+  USEMODULE += oonf_common
```

Comment:
Please review changes to NHDP and OONF in #3876 and #3882. This PR just removes the files related to `socket_base `